force-app/main/default/classes/LeadTest.cls

```diff
@@ -0,0 +1,102 @@
+        List<Lead> scoredLeads = new List<Lead>();
+        Lead lead1 = new Lead(FirstName = 'Bob', LastName = 'Smith', LeadSource = 'Web', Company = 'Test Co', Email = 'test@testing.com', Phone = '555-555-5555');
+        Lead lead2 = new Lead(FirstName = 'Joe', LastName = 'Cross', Industry = 'Technology', Company = 'Test Co', Phone = '333-333-3333');
+        Lead lead3 = new Lead(FirstName = 'Betty', LastName = 'Roberts', LeadSource = 'Other', Company = 'Test Co', Email = 'betty@boop.com');
+        Lead lead4 = new Lead(FirstName = 'Jim', LastName = 'Bailey', Title = 'exec', Company = 'Test Co');
+        scoredLeads.add(lead1);
+        scoredLeads.add(lead2);
+        scoredLeads.add(lead3);
+        scoredLeads.add(lead4);
+
+        Test.startTest();
+        Database.DMLOptions dml = new Database.DMLOptions(); 
+        dml.DuplicateRuleHeader.allowSave = true;
+        dml.DuplicateRuleHeader.runAsCurrentUser = true;
+        Database.insert(scoredLeads, dml);
+        System.debug('Test = New lead1 id is ' + lead1.Id);
+        Test.stopTest();
+        
+        Lead updatedLead1 = [SELECT Name, Id, Lead_Score__c FROM Lead WHERE Id = :lead1.Id];
+        Lead updatedLead2 = [SELECT Name, Id, Lead_Score__c FROM Lead WHERE Id = :lead2.Id];
+        Lead updatedLead3 = [SELECT Name, Id, Lead_Score__c FROM Lead WHERE Id = :lead3.Id];
+        Lead updatedLead4 = [SELECT Name, Id, Lead_Score__c FROM Lead WHERE Id = :lead4.Id];
+
+        Assert.areEqual (8, updatedLead1.Lead_score__c, 'Lead score is not calculated properly');
+        Assert.areEqual(15, updatedLead2.Lead_score__c, 'Lead score is not calculated properly');
+        Assert.areEqual(0, updatedLead3.Lead_score__c, 'Lead score is not calculated properly');
+        Assert.areEqual(0, updatedLead4.Lead_score__c, 'Lead score is not calculated properly');
+
+        lead4.industry = 'Technology';
+        update lead4;
+        Lead updatedLead = [SELECT Lead_Score__c FROM Lead WHERE Id = :lead4.Id];
+        Assert.areEqual (10, updatedLead.Lead_Score__c, 'Lead score is not calculated properly');
+    }
+
+    @isTest
+    public static void testLeadTriggerHandler_handleLeadAutoConvert(){
+        Account testAccount = new Account(Name = 'Test');
+        Insert testAccount;
+
+        List<Contact> existingContacts = new List<Contact>();
+        Contact con1 = new Contact(FirstName = 'Bob', LastName = 'Smith', Email = 'bob@testco.com', AccountId = testAccount.Id);
+        existingContacts.add(con1);
+
+        List<Lead> convertLeads = new List<Lead>();
+        Lead lead1 = new Lead(FirstName = 'Bob', LastName = 'Smith', LeadSource = 'Web', Company = 'Test Co', Phone = '555-555-5555');
+        Lead lead2 = new Lead(FirstName = 'Joe', LastName = 'Cross', Industry = 'Technology', Company = 'Test Co', Phone = '333-333-3333');
+        Lead lead3 = new Lead(FirstName = 'Betty', LastName = 'Roberts', LeadSource = 'Other', Company = 'Test Co', Email = 'betty@boop.com');
```

Comment:
A best practice is to test with at least 400 records so that your trigger fire multiple times.